sos/report/plugins/mellanox_firmware.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2023 Nvidia Corporation, Alin Serdean <aserdean@nvidia.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin
+import os
+import time
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
+
+    short_desc = 'Nvidia(Mellanox) firmware tools output'
+
+    plugin_name = "mellanox_firmware"
+    profiles = ('hardware', 'system')
+    packages = ('mst', 'mstflint')
+
+    MLNX_STRING = "Mellanox Technologies"
+
+    def check_enabled(self):
+        """
+        Checks if this plugin should be executed at all.
+        We will only enable the plugin if there is a
+        Mellanox Technologies network adapter
+        """
+        lspci = self.exec_cmd("lspci -D -d 15b3::0200")
+        return lspci['status'] == 0 and self.MLNX_STRING in lspci['output']
+
+    def collect(self):
+        if not self.get_option('allow_system_changes'):
+            self._log_info("Skipping to collect plugin "
+                           f"{self.plugin_name} as it requires "
+                           "--allow-system-changes option.")
```

Comment:
Let's reword this, as it makes it sound like the plugin does nothing without this option.

Perhaps something like `Skipping mst/mlx cable commands as system changes would be made. Use --allow-system-changes to enable this collection`?

Reply:
Thank you for the suggestion. I folded in the changes.